sourcegraph/sourcegraph
# use vite for web builds

[Vite](https://vitejs.dev/) is a fast web bundler. With Vite, a local dev server is available as fast as with esbuild, and incremental builds are (1) hot (no page reload needed) and (2) much faster (<500ms).

## Test plan

CI for e2e tests, and manually testing the dev experience:

1. Run `sg start` or whatever (I use `sg start single-program`) to get a local dev server
2. Go to https://sourcegraph.test:3443/search
3. Add some text to GlobalNavbar.tsx like `asdf1234` and hit save in your editor
4. Confirm that the edit is reflected immediately in your web browser without needing a full page reload
